From d07676efaa48a5f0f154c845d649ccf111d07bc7 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Sat, 25 Apr 2020 05:19:40 +0800
Subject: [PATCH 35/37] MLK-23846 ARM64: dts: freescale: fsl-imx8mn-ddr4-evk:
 correct bd71847 device node

commit 8507afc3a3973646ab198d29429350b44ddda699 from
https://source.codeaurora.org/external/imx/linux-imx.git

The legacy bd71847 dts cherry-pick from i.mx_4.14 is not suitable for
i.mx_4.19 because upstreaming bd71847 requlator used on 4.19, so the dts
should be updated too.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Jacky Bai <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8mn-ddr4-evk.dts     | 100 +++++-------------
 1 file changed, 29 insertions(+), 71 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts
index 100cb502e0d7..b9f80a784c55 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mn-ddr4-evk.dts
@@ -512,27 +512,22 @@
 	sda-gpios = <&gpio5 15 GPIO_ACTIVE_HIGH>;
 	status = "okay";
 
-	pmic: bd71837@4b {
+	pmic@4b {
 		reg = <0x4b>;
-		compatible = "rohm,bd71837";
+		compatible = "rohm,bd71847";
 		/* PMIC BD71837 PMIC_nINT GPIO1_IO3 */
 		pinctrl-0 = <&pinctrl_pmic>;
-		gpio_intr = <&gpio1 3 GPIO_ACTIVE_LOW>;
-
-		gpo {
-			rohm,drv = <0x0C>;	/* 0b0000_1100 all gpos with cmos output mode */
-		};
+		interrupt-parent = <&gpio1>;
+		interrupts = <3 GPIO_ACTIVE_LOW>;
+		rohm,reset-snvs-powered;
 
 		regulators {
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			bd71837,pmic-buck2-uses-i2c-dvs;
-			bd71837,pmic-buck2-dvs-voltage = <1000000>, <900000>, <0>; /* VDD_ARM: Run-Idle */
-
 			buck1_reg: regulator@0 {
 				reg = <0>;
-				regulator-compatible = "buck1";
+				regulator-compatible = "BUCK1";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1300000>;
 				regulator-boot-on;
@@ -542,118 +537,81 @@
 
 			buck2_reg: regulator@1 {
 				reg = <1>;
-				regulator-compatible = "buck2";
+				regulator-compatible = "BUCK2";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1300000>;
 				regulator-boot-on;
 				regulator-always-on;
 				regulator-ramp-delay = <1250>;
-			};
-
-			buck3_reg: regulator@2 {
-				reg = <2>;
-				regulator-compatible = "buck3";
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1300000>;
+				rohm,dvs-run-voltage = <1000000>;
+				rohm,dvs-idle-voltage = <900000>;
 			};
 
 			buck4_reg: regulator@3 {
+				// BUCK6 in datasheet
 				reg = <3>;
-				regulator-compatible = "buck4";
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1300000>;
-			};
-
-			buck5_reg: regulator@4 {
-				reg = <4>;
-				regulator-compatible = "buck5";
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1350000>;
-			};
-
-			buck6_reg: regulator@5 {
-				reg = <5>;
-				regulator-compatible = "buck6";
+				regulator-compatible = "BUCK4";
 				regulator-min-microvolt = <3000000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck7_reg: regulator@6 {
-				reg = <6>;
-				regulator-compatible = "buck7";
+			buck5_reg: regulator@4 {
+				// BUCK7 in datasheet
+				reg = <4>;
+				regulator-compatible = "BUCK5";
 				regulator-min-microvolt = <1605000>;
 				regulator-max-microvolt = <1995000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck8_reg: regulator@7 {
-				reg = <7>;
-				regulator-compatible = "buck8";
+			buck6_reg: regulator@5 {
+				// BUCK8 in datasheet
+				reg = <5>;
+				regulator-compatible = "BUCK6";
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <1400000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo1_reg: regulator@8 {
-				reg = <8>;
-				regulator-compatible = "ldo1";
+			ldo1_reg: regulator@6 {
+				reg = <6>;
+				regulator-compatible = "LDO1";
 				regulator-min-microvolt = <1600000>;
 				regulator-max-microvolt = <1900000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo2_reg: regulator@9 {
-				reg = <9>;
-				regulator-compatible = "ldo2";
+			ldo2_reg: regulator@7 {
+				reg = <7>;
+				regulator-compatible = "LDO2";
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <900000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo3_reg: regulator@10 {
-				reg = <10>;
-				regulator-compatible = "ldo3";
+			ldo3_reg: regulator@8 {
+				reg = <8>;
+				regulator-compatible = "LDO3";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo4_reg: regulator@11 {
-				reg = <11>;
-				regulator-compatible = "ldo4";
-				regulator-min-microvolt = <900000>;
-				regulator-max-microvolt = <1800000>;
-			};
-
-			ldo5_reg: regulator@12 {
-				reg = <12>;
-				regulator-compatible = "ldo5";
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-			};
-
 			ldo6_reg: regulator@13 {
 				reg = <13>;
-				regulator-compatible = "ldo6";
+				regulator-compatible = "LDO6";
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
-
-			ldo7_reg: regulator@14 {
-				reg = <14>;
-				regulator-compatible = "ldo7";
-				regulator-min-microvolt = <1800000>;
-				regulator-max-microvolt = <3300000>;
-			};
 		};
 	};
 };
-- 
2.17.1

